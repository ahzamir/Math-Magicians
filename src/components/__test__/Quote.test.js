import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

import QuotePage from '../pages/Quote';

describe('Quote Page Testing', () => {
  render(<QuotePage />);
  test('Testing the dom for success rendering elements on Quote page', () => {
    expect(screen.queryByTestId('quoteParagraph')).toBeInTheDocument();
  });

  test('Testing the dom for non existing elemente on the page after rendering', () => {
    expect(screen.queryByTestId('nothing')).not.toBeInTheDocument();
  });
});
