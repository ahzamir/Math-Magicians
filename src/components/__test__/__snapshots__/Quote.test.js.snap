// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote Page Testing renders correctly 1`] = `
<p
  className="quote border"
  data-testid="quoteParagraph"
>
  Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. -William Paul Thurston
</p>
`;
